HospitalSystem.ConsoleUI/Secretary/CrudUI.cs

```diff
@@ -0,0 +1,311 @@
+        Console.Clear();
+        UserService us = _hospital.UserService;
+        
+        System.Console.Write("Enter the user mail to view his data: ");
+        string? email = Console.ReadLine();
+        if (email is null)
+        {
+            throw new NullInputException("Null value as input");
+        }
+        var user = us.Get(email);
+        Patient pat = _hospital.PatientRepo.GetPatientById((ObjectId) user.Person.Id);
+        System.Console.WriteLine("Email : " + user.Email.ToString());
+        System.Console.WriteLine("Password : " + user.Password.ToString());
+        System.Console.WriteLine("First Name : " + pat.FirstName);
+        System.Console.WriteLine("Last Name : " + pat.LastName);
+        System.Console.WriteLine("");
+        System.Console.Write("Type back to get to menu: ");
+        string? back = Console.ReadLine();
+
+        while(back != "back"){
+        Console.Clear();
+        System.Console.Write("Type back to get to menu: ");
+        back = Console.ReadLine();
+        }
+
+        Console.Clear();
+    }
+
+    public void updateUserPatient()
+    {
+        Console.Clear();
+        UserService us = _hospital.UserService;
+        System.Console.Write("Enter <email> or <password> depending of what you want update: ");
+        
+        string? enter = Console.ReadLine();
+        if(enter == "email"){
+            System.Console.Write("Enter users email: ");
+            string? email = Console.ReadLine();
+            if (email is null)
+            {
+                throw new NullInputException("Null value as input");
+            }
+            System.Console.Write("Enter new email: ");
+            string? emailNew = Console.ReadLine();
+            if (emailNew is null)
+            {
+                throw new NullInputException("Null value as input");
+            }
+            us.UpdateEmail(email,emailNew);
+        }
+        else if(enter == "password"){
+            System.Console.Write("Enter users password : ");
+            string? email = Console.ReadLine();
+            if (email is null)
+            {
+                throw new NullInputException("Null value as input");
+            }
+            System.Console.Write("Enter new password: ");
+            string? passwordNew = Console.ReadLine();
+            if (passwordNew is null)
+            {
+                throw new NullInputException("Null value as input");
+            }
+            us.UpdatePassword(email,passwordNew);
+        }
+        Console.Clear();
+    }
+
+    public void DeleteUserPatient()
+    {
+        Console.Clear();
+        UserService us = _hospital.UserService;
+        System.Console.Write("Enter the user mail to delete: ");
+        string? email = Console.ReadLine();
+        if (email is null)
+        {
+            throw new NullInputException("Null value as input");
+        }
+        us.Delete(email);
+        Console.Clear();
+    }
+
+    public void BlockUserPatients()
+    {
```

Comment:
You are only blocking one Patient. Also maybe shouldn't call it UserPatient

Reply:
Why is this function called block if it deletes? The code seems to be for blocking??